sqs/src/main/scala/akka/stream/alpakka/sqs/javadsl/SqsFlow.scala

```diff
@@ -11,49 +11,55 @@ import akka.stream.javadsl.Flow
 import akka.stream.scaladsl.{Flow => SFlow}
 import com.amazonaws.services.sqs.AmazonSQSAsync
 import java.lang.{Iterable => JIterable}
+
+import com.amazonaws.services.sqs.model.SendMessageRequest
+
 import scala.collection.JavaConverters._
 
 object SqsFlow {
 
   /**
    * Java API: creates a flow based on [[SqsFlowStage]] for a SQS queue using an [[AmazonSQSAsync]]
    */
-  def create(queueUrl: String, settings: SqsSinkSettings, sqsClient: AmazonSQSAsync): Flow[String, Result, NotUsed] =
+  def create(queueUrl: String,
+             settings: SqsSinkSettings,
+             sqsClient: AmazonSQSAsync): Flow[SendMessageRequest, Result, NotUsed] =
```

Comment:
Can we keep one signature that accepts strings perhaps, and just map into this one?

Reply:
The String based flow would require additional data to add the attributes so it would need to add those to the flow somehow. This would be possible by adding a second flow stage that maps the request to request with provided attributes, but the attribute set would need to be added to the signature, adding another 6 methods to the java api to support attributes with and without settings.

Reply:
I could create a `simple(queueUrl: String, settings: SqsSinkSettings, sqsClient: AmazonSQSAsync): Flow[String, Result, Future[Done]]` maybe?

Reply:
Why couldn't it just do `.map(new SendMessageRequest(queueUrl, _)).via(...)`, isn't that exactly what the original implementation did?

Reply:
The original created and sent the `SendMessageRequest` inside the `SqsFlowStage` which was why I couldn't just add a new part to do the conversion.

It would be simple to add for the `ScalaDsl` but as I mentioned the `JavaDsl` gets very unwieldy if we want to support the various parameter combinations.

Reply:
Ah, ok, fair enough skip it then, it is easy enough for users to create themselves even if it had been nice not to break existing code

Reply:
I agree, I couldn't find a sane way to not break old code without totally reimplementing the internals of the flow, so thought this was the more suitable and future flexible solution.